src/Broadcast.chpl

```diff
@@ -135,4 +145,42 @@ module Broadcast {
     expandedVals = (+ scan expandedVals);
     return (expandedVals == 1);
   }
+
+  proc broadcast(segs: [?sD] int, segString: borrowed SegString, size: int) throws {
+    ref segOff = segString.offsets.a;
+    const high = sD.high;
+    var strSize: int;
+    var diffs: [sD] int;
+
+    forall (i, d, seg, off) in zip (sD, diffs, segs, segOff) with (+ reduce strSize) {
+      if i == high {
+        strSize += (size - seg) * (segString.nBytes - off);
+        d = segString.nBytes - off;
+      } else {
+        strSize += (segs[i+1] - seg) * (segOff[i+1] - off);
+        d = segOff[i+1] - off;
+      }
+    }
+
+    var broadDist = broadcast(segs, diffs, size);
+    var offDiff = makeDistArray(size, int);
+    offDiff[1..] = broadDist[..size-2];
+
+    var offsets = (+ scan offDiff);
+    var r: [sD] int = 0..segString.size; 
+    var ind = broadcast(segs, r, size);
+    var expandedVals = makeDistArray(strSize, uint(8));
+    var agg = newDstAggregator(uint(8));
+    ref vals = segString.values.a;
+
+    forall (i, o, s) in zip(ind, offsets, 0..#size) with (ref agg, ref vals) {
+      var inds = if i == high then segOff[i]..segString.nBytes-1 else segOff[i]..segOff[i+1]-1;
+      var offs = if s == size - 1 then o..strSize-1 else o..offsets[s+1]-1;
+      forall (off, idx) in zip(offs, inds) with (ref agg, ref vals) {
+        agg.copy(expandedVals[off], vals[idx]);
+      }
```

Comment:
im a bit worried it might be possible to have remote to remote aggregation here. I pulled down the PR and ran a string broadcast with gasnet and 2 locales. I'm not sure if this is the cause, but I did get a server crash with a `SIGSEGV(11)`
```python
>>> s = ak.array(["abc","abcde","ab"])
>>> segs = ak.array([0,2,5])
>>> ak.groupbyclass.broadcast(segs, s, 8)
```
from the logs
```
*** Caught a fatal signal (proc 1): SIGSEGV(11)
*** NOTICE (proc 1): Before reporting bugs, run with GASNET_BACKTRACE=1 in the environment to generate a backtrace.
*** NOTICE (proc 1): We recommend linking the debug version of GASNet to assist you in resolving this application issue.
*** Caught a fatal signal (proc 1): SIGSEGV(11)
````

Reply:
The `agg` was the cause of this, saying the operation hit the memory limit. I reverted this to use the original bulk copy method, which shows equivalent times at `10**6` problem size. I had no problem running the bulk copy version with multiple locales